src/main/java/com/pracore/user/services/UserService.java

```diff
@@ -45,4 +50,69 @@ public boolean delete(int id) {
         return true;
     }
 
+    /**
+     * /propagation/required
+     */
+    @Transactional
+    public void transaction1() {
+        this.createUser("1");
+        hobbyService.transaction1();
+    }
+
+    /**
+     * /propagation/supported
+     */
+    @Transactional
+    public void transaction2() {
+        this.createUser("2");
+        hobbyService.transaction2();
+    }
+
+    /**
+     * /propagation/unsupported
+     */
+    @Transactional
+    public void transaction3() {
+        this.createUser("3");
+        hobbyService.transaction3();
+        throw new RuntimeException();
+    }
+
+
+
+    /**
+     * /propagation/requiredNew
+     */
+    @Transactional
+    public void transaction4() {
+        this.createUser("4");
+        hobbyService.transaction4();
+        throw new RuntimeException();
+    }
+
+    /**
+     * /propagation/mandatory
+     */
+
+    public void transaction5() {
+        this.createUser("5");
+        hobbyService.transaction5();
+    }
+
+    /**
+     * /propagation/never
+     */
+    @Transactional(propagation = Propagation.NEVER)
+    public void transaction6() {
```

Comment:
Can we query db and make sure transaction is role back ?

Reply:
We checked it by manually going to database and checked records. Failure transaction got rolled back